akka-http-core/src/test/scala/akka/http/impl/engine/client/ConnectionPoolSpec.scala

```diff
@@ -409,6 +411,61 @@ abstract class ConnectionPoolSpec(poolImplementation: PoolImplementation) extend
       def issueRequest(request: HttpRequest, settings: ConnectionPoolSettings): Future[HttpResponse] =
         Http().singleRequest(request, settings = settings)
     }
+
+    "support receiving a response before the request is complete" in new LocalTestSetup {
+      val sourceQueuePromise = Promise[SourceQueueWithComplete[ChunkStreamPart]]()
+      val source = Source.queue(8, OverflowStrategy.fail).mapMaterializedValue(sourceQueuePromise.success)
+      val slowEntity = Chunked(ContentTypes.`text/plain(UTF-8)`, source)
+      val request = HttpRequest(uri = s"http://$serverHostName:$serverPort/abc?query#fragment", entity = slowEntity)
+      val responseFuture = Http().singleRequest(request)
+      val sourceQueue = Await.result(sourceQueuePromise.future, 3.seconds)
+      val response = Await.result(responseFuture, 1.second.dilated)
+      response.headers should contain(RawHeader("Req-Host", s"$serverHostName:$serverPort"))
+
+      sourceQueue.offer(HttpEntity.Chunk("lala")).futureValue should be(QueueOfferResult.Enqueued)
+      sourceQueue.offer(LastChunk).futureValue should be(QueueOfferResult.Enqueued)
+      sourceQueue.complete()
+      val bytes = response.entity.dataBytes.runReduce(_ ++ _)
+      Await.result(bytes, 3.seconds) should be(ByteString("lala"))
+    }
+
+    /**
+     * Currently failing the 'outgoing request' part of the connection may also fail the 'incoming reply' part of the connection.
+     * In the future we may want to disconnect those and allow the server we connect to to choose how to handle the failure
+     * of the request entity.
+     */
```

Comment:
Hmmm yeah I wonder about that, I guess we could but would rather have someone request for it specifically

Reply:
I tried it for a bit but it didn't seem straightforward, so I agree we should probably leave it be until we find a strong use case for it.